tree/tree/inc/TLeaf.h

```diff
@@ -18,8 +18,23 @@
 // TLeaf                                                                //
 //                                                                      //
 // A TTree object is a list of TBranch.                                 //
-// A TBranch object is a list of TLeaf.                                 //
-// A TLeaf describes the branch data types.                             //
+// A TBranch object is a list of TLeaf.  In most cases, the TBranch     //
+// will have one TLeaf.                                                 //
+// A TLeaf describes the branch data types and holds the data.          //
+//                                                                      //
+// A few notes about the data held by the leaf.  It can contain:        //
+//   1 a single object or primitive (e.g., one float),                  //
+//   2 a fixed-number of objects (e.g., each entry has two floats).     //
+//     The number of elements per entry is saved in `fLen`.             //
+//   3 a dynamic number of primitives.  The number of objects in each   //
+//     entry is saved in the `fLeafCount` branch.                       //
+//                                                                      //
+// Note options (2) and (3) can combined - if fLeafCount says an entry  //
+// has 3 elements and fLen is 2, then there will be 6 objects in that   //
+// entry.                                                               //
+//                                                                      //
+// Additionally, `fNdata` is transient and generated on read to         //
+// determine the necessary size of a buffer to hold event data.         //
```

Comment:
>  +// determine the necessary size of a buffer to hold event data.         // 

This is not quite accurate.
```
fNdata == fLen * Int_t(fLeafCount->GetValue());
```
while the 'the necessary size of a buffer to hold event data' sounds more like
```
fLen * fLeafCount->GetMaximum();
```

Reply:
Well, it does both.  Here's where it's used for your second example:

https://github.com/root-project/root/blob/master/tree/tree/src/TLeaf.cxx#L350